lib/rumoji/emoji.rb

```diff
@@ -53,8 +57,14 @@ def multiple?
 
     ALL_REGEXP = Regexp.new(ALL.map(&:string).join('|'))
 
+    SYMBOL_LOOKUP = ALL.each.with_object({}) do |emoji, lookup|
+      emoji.symbols.each do |symbol|
+        lookup[symbol.to_s] = emoji
+      end
+    end
+
     def self.find(symbol)
-      ALL.find {|emoji| emoji.include? symbol }
+      SYMBOL_LOOKUP[symbol.to_s]
```

Comment:
As far as `Rumoji.decode` goes, the revised version will always be passing a String into `Emoji.find`, so there is no new String created there (`String#to_s` simply returns `self`). Other users of this gem _might_ be calling `Emoji.find` directly and passing in symbols, but I don't think it's necessary to optimize for that possibility.